torch/csrc/jit/codegen/cuda/lower_utils.h

```diff
@@ -55,6 +56,30 @@ Expr* firstInnerMostScope(Expr* scope);
 
 namespace ir_utils {
 
+// Somtimes we want to temporarily view a tensorview with another tensordomain.
+// This isn't a permanent transformation, but in indexing we want to index
+// producers with a consumer set of indices, so we need to view the producer
+// transformed like consumer while we index. This will set the tv with td for
+// the life of this context guard.
+class TVDomainGuard {
+ public:
+  TensorView* tv_;
```

Comment:
1. if public, no need for _ suffix
2. = nullptr (same for prev_domain)

Reply:
Should be private, thanks.